.eslintrc.js

```diff
@@ -0,0 +1,33 @@
+module.exports = {
+  env: {
+    node: true,
+    es2022: true,
+    browser: true,
+  },
+  extends: ['eslint:recommended', 'plugin:astro/recommended'],
+  parserOptions: {
+    ecmaVersion: 'latest',
+    sourceType: 'module',
+  },
+  rules: {},
+  overrides: [
+    {
+      files: ["*.astro"],
+      parser: "astro-eslint-parser",
+      parserOptions: {
+        parser: "@typescript-eslint/parser",
+        extraFileExtensions: [".astro"],
+      },
+      rules: {},
+    },
```

Comment:
Just a couple of questions about the eslint plugin.

Is it possible for `plugin:astro/recommended` to set this for us? Or can overrides like this only be set by a user?

And do you always need `@typescript-eslint` for Astro files? If so, maybe that could be a dependency of `astro-eslint`?

Reply:
This parser configuration is actually already included in `"plugin:astro/recommended"`. If we do not change the current configuration, we may not need to add the parser configuration to `.eslintrc.*`.
However, the parser configuration is cannot merged with other configurations, so if we don't explicitly configure it in `overrides`, a simple configuration change will break it. For example, adding a shared configuration with a parser configuration to `extends` will prevent eslint from parsing `.astro`.
Therefore, I recommend explicitly configuring the parser configs for `overrides`.

eslint-plugin-astro depends on `@typescript-eslint/parser` as peerDeps. I think this is necessary because we can always write TypeScript in `.astro`. Are there many astro users who don't actually use TypeScript? In that case, we may need to consider changing it so that it doesn't depend on `@typescript-eslint/parser`.

Reply:
Thanks for the clarification. I think there are definitely people who don’t write TypeScript with Astro (or don’t _know_ they are writing TypeScript!) I don’t think you need to stop depending on `@typescript-eslint/parser` though, the easier thing for people less familiar with TS would be if they didn’t have to manually install `@typescript-eslint`, so moving it from `peerDependencies` to `dependencies` seems like a good idea to me.

But we can continue this discussion on your repo. For this PR, this looks good to me!

Reply:
That's common argument in the eslint plugin community 😅 . If we use `dependencies`, they may be placed deep in `node_modules` and not use a version of the parser that you did not intend. Therefore, it is safer to make it `peerDependencies`.
Even now, eslint-plugin-vue still depends on vue-eslint-parser with `dependencies`, but users often complain that an unintended version is used :sweat:.

Reply:
😅! In general I prefer to use dependencies to simplify installation except for clear cases where the dependency is a bigger part of the user’s usage. So Astro or React would be peer deps because they’re the big important piece, and in this case, `eslint` should be a peer dep, but specific parsers seem safe in dependencies. I’ll trust you on this though!